msteams-platform/concepts/deploy-and-publish/appsource/prepare/frequently-failed-cases.md

```diff
@@ -1,246 +1,348 @@
 
-![OneNote allows users to paste a OneNote link in case notes can not be found](../../../../assets/images/faq/tab-onenote-config.png)
+> [!TIP]
+> If you want information on creating a high-quality experience, see the [bot design guidelines](~/bots/design/bots.md).
+
+### 6.1 Bot commands
 
-![Users can always create new plan on planner in case there no existing ones](../../../../assets/images/faq/tab-planner-config.png)
+* All commands that your bot supports must work correctly, including the "Hi", "Hello", and "Help" command.
+* Supported bot commands must be listed in the app manifest.
+* Use command lists. Analyzing user input or predicting user intent is hard. Instead of letting users guess what your bot can do, provide a list of commands your bot understands.
 
-![SharePoint also allows user to directly paste a SharePoint link](../../../../assets/images/faq/tab-sp-config.png)
+### 6.2 Bot welcome messages
 
-### &#9989; Tabs in channel - Member access
+* Bots should almost always send first-run welcome messages. For the best experience, the message should include the value proposition of your bot, how to configure the bot, and briefly describe all supported bot commands. You can display the messaging using an Adaptive Card with buttons for better usability. For more information, see [how to trigger a bot welcome message](~/bots/how-to/conversations/send-proactive-messages.md).
+* Channel bot welcome messages are optional during first run, especially if bot is available in personal scope performing similar actions. This bot must not send a personal message to all members of the team or channel as that will be considered as spamming. The welcome message should also mention the person who installed/added the bot to the channel.
+* Notification-only bots must send a welcome message that includes a message conveying, "I am a notification-only bot and will not be able to reply to your chats".
 
-* A tab configured by a member in a channel scope must be accessible to the other members without having to seek permissions from the member who configured the tab.
-* The app must provide the permission management options upfront if the tab is for private or restricted use or requires any permissions from the member who configured the tab.
+### 6.3 Bot message spamming
 
-### &#9989; Bots must always be responsive and fail gracefully
+Bots must not spam users by sending multiple messages in short succession.
 
-Your bot must be responsive to any command and not dead-end the user. Here are some tips to help your bot intelligently respond to users:
+* **Bot channel messages**: Don't spam users by creating separate new chat posts. Create a single thread post with replies in the same thread.
+* **Bot personal messages**: Don't send multiple messages. Send one message with complete information. Repeating the same welcome message over regular intervals is not allowed and is considered spamming.
```

Comment:
@heath-hamilton, please update - 
•	**Bot personal messages:** Don't send multiple messages. Send one message with complete information. Avoid multiturn conversations to complete a single workflow. Consider using task module/form to collect all inputs from user in one go instead of multiple questions. 
•	**Welcome message spamming:** 
o	Repeating the same welcome message over regular intervals is not allowed and is considered spamming.
o	Whenever new member in added in Teams don’t spam the team members by triggering welcome message in channel or personal scope.